internal/neofs/neofs.go

```diff
@@ -277,6 +277,15 @@ func (x *NeoFS) CreateObject(ctx context.Context, prm layer.PrmObjectCreate) (oi
 		return oid.ID{}, fmt.Errorf("save object via connection pool: %w", err)
 	}
 
+	// Lock payload has been written to header directly, thus we shouldn't miss this payload too.
+	if data := obj.Payload(); len(data) > 0 {
+		if prm.Payload != nil {
+			prm.Payload = io.MultiReader(bytes.NewReader(data), prm.Payload)
```

Comment:
I'd say that this case is an invalid use of an API. It can be checked above for locks specifically...

Reply:
Because you can't have locks and payload at the same time.

Reply:
This fact makes the code easier